.github/workflows/cron-nightly-rust.yml

```diff
@@ -0,0 +1,52 @@
+name: Update nightly Rust
+on:
+  schedule:
+    - cron: "29 17 1 * *" # At 17:29 on day-of-month 1.
+  workflow_dispatch: # allows manual triggering
+jobs:
+  format:
+    name: Update nightly Rustc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Update rust-toolchain.toml to use latest nightly
+        run: |
+          set -x
+
+          # Not every night has a nightly, so extract the date from whatever
+          # version of the compiler dtolnay/rust-toolchain gives us.
+          NIGHTLY_DATE="$(rustc +nightly --verbose --version | sed -ne 's/^commit-date: //p')"
+
+          # Update the nightly version in the rust-toolchain.toml file.
+          echo "Updating rust-toolchain.toml to use nightly-$NIGHTLY_DATE"
+          sed -i "s/^channel = \"nightly-.*\"/channel = \"nightly-$NIGHTLY_DATE\"/" rust-toolchain.toml
+
+          # Update the nightly date in the environment.
+          echo "nightly_date=${NIGHTLY_DATE}" >> "$GITHUB_ENV"
+
+          # Maybe there is no new nightly.
+          # In this case don't make an empty PR.
+          if ! git diff --exit-code > /dev/null; then
+              echo "Updated nightly. Opening PR."
+              echo "changes_made=true" >> "$GITHUB_ENV"
+          else
+              echo "Attempted to update nightly but the latest-nightly date did not change. Not opening any PR."
+              echo "changes_made=false" >> "$GITHUB_ENV"
+          fi
+      - name: Create Pull Request
+        if: env.changes_made == 'true'
+        uses: peter-evans/create-pull-request@v7
+        env:
+          NIGHTLY_DATE: ${{ env.nightly_date }}
+        with:
+          token: ${{ secrets.STOROPOLI_CREATE_PR_TOKEN }}
+          author: Update Nightly Rustc Bot <bot@example.com>
+          committer: Update Nightly Rustc Bot <bot@example.com>
+          title: Automated update to rustc (to nightly-$NIGHTLY_DATE)
+          body: |
+            Automated update to Rust nightly by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action
+          commit-message: Automated update to Rust nightly-$NIGHTLY_DATE
+          branch: create-pull-request/automated-nightly-update
```

Comment:
I need help reviewing this.
Cc @AaronFeickert and @Rajil1213

Reply:
This will likely trigger some `zizmor` lints. Would recommend running it locally as a first pass.